includes/class-amp-travel-blocks.php

```diff
@@ -0,0 +1,199 @@
+			array( 'wp-blocks' )
+		);
+	}
+
+	/**
+	 * Add the amp-specific html tags required by theme to allowed tags.
+	 *
+	 * @param array  $allowed_tags Allowed tags.
+	 * @param string $context Context.
+	 * @return array Modified tags.
+	 */
+	public function filter_wp_kses_allowed_html( $allowed_tags, $context ) {
+		if ( 'post' === $context ) {
+			$amp_tags = array(
+				'amp-img'  => array_merge( $allowed_tags['img'], array(
+					'attribution' => true,
+					'class'       => true,
+					'fallback'    => true,
+					'heights'     => true,
+					'media'       => true,
+					'noloading'   => true,
+					'on'          => true,
+					'placeholder' => true,
+					'srcset'      => true,
+					'sizes'       => true,
+					'layout'      => true,
+				) ),
+				'amp-list' => array(
+					'class'            => true,
+					'credentials'      => true,
+					'placeholder'      => true,
+					'noloading'        => true,
+					'on'               => true,
+					'items'            => true,
+					'max-items'        => true,
+					'single-item'      => true,
+					'reset-on-refresh' => true,
+					'src'              => true,
+					'[src]'            => true,
+					'width'            => true,
+					'height'           => true,
+					'layout'           => true,
+					'fallback'         => true,
+				),
+			);
+
+			$allowed_tags = array_merge( $allowed_tags, $amp_tags );
+		}
+		return $allowed_tags;
+	}
+
+	/**
+	 * Register Travel Discover block type.
+	 */
+	public function register_block_discover() {
+		register_block_type( 'amp-travel/discover', array(
+			'attributes'      => array(
+				'heading'    => array(
+					'type'    => 'string',
+					'default' => __( 'Discover Adventures', 'travel' ),
+				),
+				'subheading' => array(
+					'type'    => 'string',
+					'default' => __( 'Get inspired and find your next big trip', 'travel' ),
+				),
+			),
+			'render_callback' => array( $this, 'render_block_discover' ),
+		) );
+	}
+
+	/**
+	 * Front-side render for Travel Discover block.
+	 *
+	 * @param array $attributes Block attributes.
+	 * @return string Output.
+	 */
+	public function render_block_discover( $attributes ) {
+		global $post;
+
+		$args = array(
+			'numberposts' => 1,
+			'post_status' => 'publish',
+			'post_type'   => 'post',
+		);
+
+		if ( $post ) {
+			$args['exclude'] = array( $post->ID );
+		}
+
+		$posts = wp_get_recent_posts( $args );
+
+		$heading    = $attributes['heading'];
+		$subheading = $attributes['subheading'];
+
+		// If there's no post, return.
+		if ( empty( $posts ) ) {
+			return '';
+		} else {
+			$discover_post = $posts[0];
+			$title         = get_the_title( $discover_post['ID'] );
+			$link          = get_permalink( $discover_post['ID'] );
+
+			// Use the wp_trim_words directly since otherwise the_content will run endlessly due to wp_trim_excerpt() using the global post.
+			if ( empty( $discover_post['post_excerpt'] ) ) {
+				$excerpt_length = apply_filters( 'excerpt_length', 15 );
```

Comment:
Could you please put these comments over the `apply_filters()` lines like in [this example](https://github.com/Automattic/jetpack/blob/master/modules/theme-tools/content-options/blog-display.php#L40) from Jetpack:
```php
/** This filter is documented in wp-includes/formatting.php */
$excerpt_length = apply_filters( 'excerpt_length', 15 );
/** This filter is documented in wp-includes/formatting.php */
$excerpt_more   = apply_filters( 'excerpt_more', ' ...' );
```